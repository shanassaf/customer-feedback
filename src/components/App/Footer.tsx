import React from 'react';
import styled from 'styled-components';

const StyledFooter = styled.footer``;

export const Footer = () => (
	<StyledFooter>
		<img src="../images/checkout.jpeg" alt="Checkout logo" />
	</StyledFooter>
);
