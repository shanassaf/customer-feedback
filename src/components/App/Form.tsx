import React from 'react';
import styled from 'styled-components';

const StyledForm = styled.form``;
const StyledInput = styled.input``;
const StyledLabel = styled.label``;
const StyledContainer = styled.div``;
const StyledOption = styled.option``;
const StyledSelect = styled.select``;

export const Form = () => {
	return (
		<StyledForm>
			<StyledContainer>
				<StyledLabel>Name</StyledLabel>
				<StyledInput type="text" />
			</StyledContainer>
			<StyledContainer>
				<StyledLabel>Email</StyledLabel>
				<StyledInput type="text" />
			</StyledContainer>
			<StyledContainer>
				<StyledLabel>Rating</StyledLabel>
				<StyledSelect>
					<StyledOption value="1">1</StyledOption>
					<StyledOption value="2">2</StyledOption>
					<StyledOption value="3">3</StyledOption>
					<StyledOption value="4">4</StyledOption>
					<StyledOption value="5">5</StyledOption>
				</StyledSelect>
			</StyledContainer>
			<StyledContainer>
				<StyledInput type="submit" />
			</StyledContainer>
		</StyledForm>
	);
};
