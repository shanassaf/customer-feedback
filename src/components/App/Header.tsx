import React from 'react';
import styled from 'styled-components';

const StyledHeader = styled.header``;

export const Header = () => (
	<StyledHeader>
		<img src="../images/checkout.jpeg" alt="Checkout logo" />
	</StyledHeader>
);
