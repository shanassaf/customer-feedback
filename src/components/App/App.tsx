import React from 'react';
import styled from 'styled-components';
import { Form } from '../Form';
import { Header } from '../Header';
import { Footer } from '../Footer';
import { Feedback } from '../Feedback';
import { Chart } from '../Chart';

const StyledLayout = styled.div`
	display: grid;
	padding: 10px;
	grid-template-areas:
		'head'
		'form'
		'graph'
		'comments'
		'foot ';

	@media (min-width: 720px) {
		grid-template-columns: 1fr 1fr;
		grid-template-areas:
			'head head'
			'form graph'
			'comments comments'
			'foot foot';
	}
`;
const StyledForm = styled.div`
	grid-area: form;
	padding: 10px 20px;
	max-width: 640px;
	border: 1px solid black;
	border-radius: 4px;
`;
const StyledChart = styled.div`
	grid-area: graph;
	margin: auto;
	width: 80%;
	padding: 30px;
`;
const StyledFeedback = styled.div`
	grid-area: comments;
	border-top: 1px solid black;
	border-bottom: 1px solid black;
	padding: 10px 20px;
`;
const StyledHeader = styled.div`
	grid-area: head;
	height: 50px;
	margin-bottom: 20px;
`;
const StyledFooter = styled.div`
	grid-area: foot;
	padding: 50px 20px;
`;

export const App = () => {
	return (
		<StyledLayout>
			<StyledHeader>
				<Header />
			</StyledHeader>
			<StyledForm>
				<Form />
			</StyledForm>
			<StyledChart>
				<Chart />
			</StyledChart>
			<StyledFeedback>
				<Feedback />
			</StyledFeedback>
			<StyledFooter>
				<Footer />
			</StyledFooter>
		</StyledLayout>
	);
};
