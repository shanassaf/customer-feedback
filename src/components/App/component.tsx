import React from 'react';
import styled from 'styled-components';
import { Form } from './Form';
import { Header } from './Header';
import { Footer } from './Footer';

const StyledLayout = styled.div`
	display: grid;
	grid-template-areas:
		'head'
		'form'
		'graph'
		'comments '
		'foot ';

	@media (min-width: 720px) {
		grid-template-areas:
			'head head'
			'form graph'
			'comments comments'
			'foot foot';
	}
`;
const StyledForm = styled.div`
	grid-area: form;
	width: 100%;
`;
const StyledGraph = styled.div`
	grid-area: graph;
	width: 100%;
	height: 100px;
	background-color: yellow;
`;
const StyledComments = styled.div`
	grid-area: comments;
	width: 100%;
	height: 100px;
	background-color: purple;
`;
const StyledHeader = styled.div`
	grid-area: head;
	width: 100%;
	height: 100px;
	background-color: blue;
`;
const StyledFooter = styled.div`
	grid-area: foot;
	width: 100%;
	height: 100px;
	background-color: green;
`;

const App = () => {
	return (
		<StyledLayout>
			<StyledHeader>
				<Header />
			</StyledHeader>
			<StyledForm>
				<Form />
			</StyledForm>
			<StyledGraph>Graph</StyledGraph>
			<StyledComments>Comments</StyledComments>
			<StyledFooter>
				<Footer />
			</StyledFooter>
		</StyledLayout>
	);
};

export default App;
