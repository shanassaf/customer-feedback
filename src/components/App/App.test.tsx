import React from 'react';
import { render } from '@testing-library/react'
import App from './component';

describe('App', () => {
  test('component renders correctly', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
})