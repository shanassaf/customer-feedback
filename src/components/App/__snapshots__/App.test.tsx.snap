// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-fFSRdu hARJKV"
  >
    <div
      class="sc-hHEjAm bshMBs"
    >
      <header
        class="sc-jrsJCI cKuqIz"
      >
        <img
          alt="Checkout logo"
          height="20px"
          src="checkout.svg"
        />
      </header>
    </div>
    <div
      class="sc-bkbjAj gWfBiU"
    >
      <h1
        class="sc-eCApGN iEtHMq"
      >
        Customer Feedback Form
      </h1>
      <h2
        class="sc-jSFkmK deQnhz"
      >
        Please leave a review of our product here!
      </h2>
      <form
        class="sc-gKAblj hkzmtT"
      >
        <div
          class="sc-pNWxx bWjldX"
        >
          <label
            class="sc-fujyUd cwiAfG"
            for="name"
          >
            Name *
          </label>
          <input
            class="sc-bdnylx bzTEwn"
            name="name"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="sc-pNWxx bWjldX"
        >
          <label
            class="sc-fujyUd cwiAfG"
            for="email"
          >
            Email *
          </label>
          <input
            class="sc-bdnylx bzTEwn"
            name="email"
            required=""
            type="email"
            value=""
          />
        </div>
        <div
          class="sc-pNWxx bWjldX"
        >
          <label
            class="sc-fujyUd cwiAfG"
            for="comment"
          >
            Comment *
          </label>
          <textarea
            class="sc-hKFyIo drMVRo"
            name="comment"
            required=""
          />
        </div>
        <div
          class="sc-pNWxx bWjldX"
        >
          <label
            class="sc-fujyUd cwiAfG"
            for="rating"
          >
            Rating *
          </label>
          <select
            class="sc-gtssRu dgBQYk"
            name="rating"
            required=""
          >
            <option
              class="sc-dlnjPT"
              value="0"
            >
              0
            </option>
            <option
              class="sc-dlnjPT"
              value="1"
            >
              1
            </option>
            <option
              class="sc-dlnjPT"
              value="2"
            >
              2
            </option>
            <option
              class="sc-dlnjPT"
              value="3"
            >
              3
            </option>
            <option
              class="sc-dlnjPT"
              value="4"
            >
              4
            </option>
            <option
              class="sc-dlnjPT"
              value="5"
            >
              5
            </option>
          </select>
        </div>
        <div
          class="sc-pNWxx bWjldX"
        >
          <input
            class="sc-iCoHVE clHQPA"
            type="submit"
            value="Submit"
          />
        </div>
      </form>
    </div>
    <div
      class="sc-iemXMA dWcWZq"
    >
      <div
        class="sc-fnVYJo CvBch"
      >
        <div>
          Pie Chart
        </div>
      </div>
    </div>
    <div
      class="sc-dIvqjp iWFXBL"
    >
      <ul
        class="sc-bqGHjH gSeTRC"
      >
        <h2
          class="sc-crzoUp bNraUV"
        >
          Comments section
        </h2>
        <h2
          class="sc-dIsAE ciFvTP"
        >
          See the feedback other users have left here!
        </h2>
        <li
          class="sc-ksluoS hvBuRz"
        >
          <div
            class="sc-hBMVcZ"
          >
            Name: name
          </div>
          <div
            class="sc-hBMVcZ"
          >
            Email: email
          </div>
          <div
            class="sc-hBMVcZ"
          >
            Rating: 1
          </div>
          <div
            class="sc-hBMVcZ"
          >
            Comment: text
          </div>
        </li>
      </ul>
    </div>
    <div
      class="sc-dlMBXb ejZGQ"
    >
      <footer
        class="sc-kEqYlL jpYKBx"
      >
        Check out the repository for this webpage on
        <a
          class="sc-iqAbSa jjiDNQ"
          href="https://github.com/shanassaf/customer-feedback"
        >
           github
        </a>
      </footer>
    </div>
  </div>
</DocumentFragment>
`;
